enterprise/server/invocation_search_service/invocation_search_service.go

```diff
@@ -170,6 +170,93 @@ func (s *InvocationSearchService) QueryInvocations(ctx context.Context, req *inp
 	if statusQuery != "" {
 		q.AddWhereClause(fmt.Sprintf("(%s)", statusQuery), statusArgs...)
 	}
+}
+
+func (s *InvocationSearchService) searchInvocationsInExecutions(ctx context.Context, req *inpb.SearchInvocationRequest) (*query_builder.Query, error) {
+	if s.oh == nil {
+		return nil, status.UnimplementedError("Execution filters require an OLAP DB.")
+	}
+	q := query_builder.NewQuery(`SELECT DISTINCT invocation_uuid FROM Executions as i`)
+	addCommonWhereClauses(q, req.GetQuery())
+	for _, f := range req.GetQuery().GetFilter() {
+		if f.GetMetric().Execution == nil {
+			continue
+		}
+		str, args, err := filter.GenerateFilterStringAndArgs(f, "i.")
+		if err != nil {
+			return nil, err
+		}
+		q.AddWhereClause(str, args...)
+	}
+
+	// Set a high-ish limit here, but not the request's limit: there may be extra
+	// filters when we go to fetch the actual invocations that filter out even
+	// more results.
+	q.SetLimit(200)
+	qString, qArgs := q.Build()
+
+	if req.Sort != nil || req.PageToken != "" {
+		return nil, status.UnimplementedError("Sorting and pagination is not supported for execution searches.")
+	}
+	q.SetOrderBy("i.updated_at_usec", false)
+
+	// More stuff...
+	rows, err := s.oh.DB(ctx).Raw(qString, qArgs...).Rows()
+	if err != nil {
+		return nil, err
+	}
+	type o struct {
+		InvocationUuid string
+	}
+	var ids []string
+	for rows.Next() {
+		out := o{}
+		if err := s.oh.DB(ctx).ScanRows(rows, &out); err != nil {
+			return nil, err
+		}
+		u, err := uuid.Parse(out.InvocationUuid)
+		if err != nil {
+			return nil, err
+		}
+
+		ids = append(ids, u.String())
+	}
+
+	invocationQuery := query_builder.NewQuery("SELECT * FROM Invocations AS i")
+	invocationQuery.AddWhereClause("i.invocation_id IN ?", ids)
```

Comment:
I think invocation_id is not in clickhouse Invocations table. but only invocation_uuid.

Also when querying clickhouse Invocation table, it will be faster if provide group_id, updated_at_usec range.